lms/djangoapps/mobile_api/video_outlines/views.py

```diff
@@ -120,28 +121,10 @@ def get(self, request, course, *args, **kwargs):
         block_id = kwargs['block_id']
         lang = kwargs['lang']
 
-        usage_key = BlockUsageLocator(course.id, block_type='video', block_id=block_id)
-        video_descriptor = modulestore().get_item(usage_key)
-        feature_enabled = is_val_transcript_feature_enabled_for_course(usage_key.course_key)
         try:
-            transcripts = video_descriptor.get_transcripts_info(include_val_transcripts=feature_enabled)
-            content, filename, mimetype = video_descriptor.get_transcript(transcripts, lang=lang)
-        except (ValueError, NotFoundError):
-            # Fallback mechanism for edx-val transcripts
-            transcript = None
-            if feature_enabled:
-                transcript = get_video_transcript_content(video_descriptor.edx_video_id, lang)
-
-            if not transcript:
-                raise Http404(u'Transcript not found for {}, lang: {}'.format(block_id, lang))
-
-            transcript_conversion_props = dict(transcript, output_format=Transcript.SRT)
-            transcript = convert_video_transcript(**transcript_conversion_props)
-            filename = transcript['filename']
-            content = transcript['content']
-            mimetype = Transcript.mime_types[Transcript.SRT]
-        except KeyError:
-            raise Http404(u"Transcript not found for {}, lang: {}".format(block_id, lang))
+            content, filename, mimetype = get_transcript(course.id, block_id, lang)
```

Comment:
@muhammad-ammar this will also need rebasing.

Reply:
will update this once https://github.com/edx/edx-platform/pull/17549 is finalised and merged